types/ember/index.d.ts

```diff
@@ -2111,6 +2111,10 @@ declare module 'ember' {
              * supplied callback function using `this.route`.
              */
             static map(callback: (this: RouterDSL) => void): void;
+            /**
+             * This property represent the logical name of the route, which is dot separated.
+             */
+            currentRouteName: string;
```

Comment:
Can we have this as a readonly property, in alignment with the implementation

https://github.com/emberjs/ember.js/blob/b08b93c925bbc7fd4c39ad94cc8280abc56946e0/packages/ember-routing/lib/services/router.js#L41